Modules/TestQuestionPool/classes/class.ilObjQuestionPoolGUI.php

```diff
@@ -1694,13 +1694,15 @@ public static function _goto($a_target)
         $ilAccess = $DIC['ilAccess'];
         $ilErr = $DIC['ilErr'];
         $lng = $DIC['lng'];
-
-        if ($ilAccess->checkAccess("write", "", $a_target) || $ilAccess->checkAccess('read', '', $a_target)) {
-            $_GET['cmdClass'] = 'ilObjQuestionPoolGUI';
-            $_GET['cmd'] = 'questions';
-            $_GET['baseClass'] = 'ilRepositoryGUI';
-            $_GET["ref_id"] = $a_target;
-            include_once("ilias.php");
+        $ctrl = $DIC['ilCtrl'];
+
+        if ($ilAccess->checkAccess("write", "", (int) $a_target)
+            || $ilAccess->checkAccess('read', '', (int) $a_target)
+        ) {
+            $target_class = 'ilObjQuestionPoolGUI';
```

Comment:
@mbecker-databay / @nhaagen Just one thought: I personally would use `class-strings` (e.g.  `ilObjQuestionPoolGUI::class` here. This makes it easier for maintainers of these classes to change the dependent code if the class name changes (at least when refactoring the code with IDEs like `PHPStorm`).